aodndata/aatams/aatams_sattag_nrt.py

```diff
@@ -0,0 +1,111 @@
+
+logger = logging.getLogger(__name__)
+
+# AATAMS NRT global constants
+AATAMS_SATTAG_QC_NRT_BASE = "IMOS/AATAMS/AATAMS_SATTAG_QC_NRT"
+
+
+def metadata_index(file_list):
+    """Locate the metadata.csv file in a multi value PipelineFile object."""
+    fname = [x for x in file_list if "metadata" in x.name]
+    return list(file_list).index(fname[0])
+
+
+def dest_path(filepath):
+    return os.path.join(AATAMS_SATTAG_QC_NRT_BASE, os.path.basename(filepath))
+
+
+class AatamsSattagQcNRTHandler(HandlerBase):
+    """The class for aatams sattag qc NRT."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the class by only accepting zip files."""
+        super(AatamsSattagQcNRTHandler, self).__init__(*args, **kwargs)
+        self.allowed_extensions = [".zip"]
+        self.allowed_regexes = ["^.+_nrt\\.zip$"]
+        self.allowed_archive_path_regexes = [AATAMS_SATTAG_QC_NRT_BASE + ".+\\.zip"]
+        self.allowed_dest_path_regexes = [AATAMS_SATTAG_QC_NRT_BASE + ".+\\.csv"]
+        self.dest_path_function = dest_path
+        self.archive_input_file = True
+        self.archive_path_function = self.dest_path_function
+
+        # Initialize the schema class and set default validation method
+        self.schema = AatamsSattagQcDmSchema()  # Schema is the same as dm
+        self.validation_call = self.schema.extensive_validation
+        # Use below to just validate the file names and csv headers
+        # self.validation_call = self.SCHEMA.quick_validation
+        self.previous_files = None
+        self.broker = get_storage_broker(
+            self.config.pipeline_config["global"]["upload_uri"],
+        )
+
+    def is_update_required(self, old_metadata_file):
+        """Check if updates to the dataset/archive are required by inspecting the most
+        recent dates between the new stream and remote(current) metadata state."""
+        new_valid_data = self.schema.metadata
+        new_recent_date = sorted(new_valid_data["qc_end_date"])[-1]
+
+        _, old_valid_data = self.schema.validate_file(old_metadata_file)
+        old_recent_date = sorted(old_valid_data["qc_end_date"])[-1]
+        return new_recent_date > old_recent_date
+
+    def preprocess(self):
+        """Validate individual files within the zip."""
+        files_in_zip = self.file_collection.get_attribute_list("local_path")
+        self.validation_call(files_in_zip)
+
+        self.input_file_object.publish_type = PipelineFilePublishType.ARCHIVE_ONLY
+
+        self.file_collection.filter_by_attribute_value(
+            "file_type", FileType.CSV,
+        ).set_publish_types(PipelineFilePublishType.HARVEST_UPLOAD)
+
+    def process(self):
+        """Process NRT files, removing the previous files if content of new file is
+        updated."""
+
+        self.previous_files = self.state_query.query_storage(dest_path(""))
+        if self.previous_files:
+            old_metadata_file = self.previous_files[metadata_index(self.previous_files)]
```

Comment:
I found this rather confusing... why have a function return the _index_ of an item in the list, when all you want is the item itself? i.e. instead of `metadata_index` you could have
```python
def get_metadata_file(file_list):
￼    fname = [x for x in file_list if "metadata" in x.name]
￼    return fname[0]
```
And then here it would be 
```python
    old_metadata_file = get_metadata_file(self.previous_files)
    new_metadata_file = get_metadata_file(self.file_collection)
```

Reply:
Yeap. I think the lower-level functionality of matlab is getting in my way when writing python!